web-ui/src/pages/CheckinsReportPage.jsx

```diff
@@ -24,8 +26,34 @@ const CheckinsReportPage = () => {
     const bPieces = b.name.split(' ').slice(-1);
     return aPieces.toString().localeCompare(bPieces);
   });
+
   const [filteredPdls, setFilteredPdls] = useState(pdls);
 
+  const processedQPs = useRef(false);
+  useQueryParameters(
+    [
+      {
+        name: 'pdls',
+        default: [],
+        value: selectedPdls,
+        setter(ids) {
+          const newPdls = ids.map(id => pdls.find(pdl => pdl.id === id));
+          setSelectedPdls(newPdls);
+        },
+        toQP(newPdls) {
+          if (isArrayPresent(newPdls)) {
+            const ids = newPdls.map(pdl => pdl.id);
+            return ids.join(',');
+          } else {
+            return [];
+          }
+        }
+      }
+    ],
+    [pdls],
```

Comment:
The requirements for processing query parameters for this page are to have entries in the `pdl` array.